cpp/src/io/comp/gpuinflate.cu

```diff
@@ -1208,41 +1211,83 @@ CUDF_KERNEL void __launch_bounds__(1024)
   if (t < len) { dst[t] = src[t]; }
 }
 
-void gpuinflate(device_span<device_span<uint8_t const> const> inputs,
-                device_span<device_span<uint8_t> const> outputs,
-                device_span<codec_exec_result> results,
-                gzip_header_included parse_hdr,
-                rmm::cuda_stream_view stream)
-{
-  constexpr int block_size = 128;  // Threads per block
-  if (inputs.size() > 0) {
-    inflate_kernel<block_size>
-      <<<inputs.size(), block_size, 0, stream.value()>>>(inputs, outputs, results, parse_hdr);
+enum class task_type { DECOMPRESSION, COMPRESSION };
+
+class cost_model {
```

Comment:
I like the use of cost model as we can always only update this and both sort_tasks and split_tasks would update themselves